cli_tools/google-osconfig-agent/main.go

```diff
@@ -60,46 +90,33 @@ func main() {
 	}
 
 	logger.Init(ctx, proj)
+	defer logger.Close()
 
-	action := flag.Arg(0)
-	if action == "inventory" {
+	switch action := flag.Arg(0); action {
+	case "":
+		if err := service.Register(ctx, "google_osconfig_agent", "Google OSConfig Agent", "", run, "run"); err != nil {
+			logger.Fatalf("service.Register error: %v", err)
+		}
+	case "noservice":
+		run(ctx)
+	case "inventory":
 		// Just run inventory and exit.
 		inventory.RunInventory()
-		logger.Close()
-		os.Exit(0)
-	}
-
-	if action == "ospackage" {
+		return
+	case "ospackage":
 		// Just run SetConfig and exit.
```

Comment:
comment is no longer true, you're calling ospackage.Run instead of osconfig.SetConfig

Reply:
Removed comment as I think this code is pretty clear now